### PR TITLE
Configuration

### PR DESCRIPTION
**Pull request by lzukowski**

### CODE REVIEW

File: event_sourcery_django/__init__.py [hunk 1]
```diff
@@ -22,8 +24,15 @@
 from event_sourcery.event_store.outbox import Outbox
 
 
+class Config(BaseModel):
+    model_config = ConfigDict(extra="forbid", frozen=True)
```
Comment (luke9642) [inline event_sourcery_django/__init__.py:28]: What about creating something like global `ImmutableBaseModel` class?

File: event_sourcery_esdb/__init__.py [hunk 2]
```diff
@@ -27,17 +30,31 @@
...
+
+    timeout: Seconds | None = None
+    outbox_name: str = "pyes-outbox"
+    outbox_attempts: PositiveInt = 3
+
 
 @dataclass(repr=False)
 class ESDBBackendFactory(BackendFactory):
     esdb_client: EventStoreDBClient
+    config: Config = field(default_factory=Config)
```
Comment (luke9642) [inline event_sourcery_esdb/__init__.py:47]: Maybe protected field?

```suggestion
    _config: Config = field(default_factory=Config)
```

Comment (lzukowski) [reply event_sourcery_esdb/__init__.py:47]: I'm not sure. This is config which is immutable. I don't mind exposing it.

File: event_sourcery_django/__init__.py (hunk 1 above)
Comment (lzukowski) [reply event_sourcery_django/__init__.py:28]: I'm not sure. Creating a model, that in the long term can grow with some addons, or having a config. Imo config is short and explicit.

**Pull request merged**